[wasm] Misc build+test fixes

radical opened the pull request:
- [wasm] Use `-s EXPORT_ES6=1` only for link time
- [wasm] Fix loading `dotnet.native.js.symbols`
- WBT: Re-enable previously disabled tests
- Add make target `build-packages` to build the nugets
- Browser.targets - normalize tmp webcil paths

radical committed:
[wasm] Fix loading `dotnet.native.js.symbols`

radical committed:
[wasm] Use -s EXPORT_ES6=1 only for link time

Fixes `emcc: warning: linker setting ignored during compilation: 'EXPORT_ES6' [-Wunused-command-line-argument]`

radical committed:
[wasm] WBT: Re-enable skiasharp test

.. as a new build of skiasharp native lib is now available.

Fixes issue: https://github.com/dotnet/runtime/issues/82725

radical commented:
/azp run runtime-wasm

radical committed:
[wasm] WBT: Enable more previously disabled tests

`Blazor.BuildPublishTests.WithDllImportInMainAssembly`
`Blazor.BuildPublishTests.BlazorPublishRun`

Issue: https://github.com/dotnet/runtime/issues/82725
Issue: https://github.com/dotnet/runtime/issues/82481

radical committed:
[wasm] Browser.targets - fix up paths for webcil

radical committed:
[wasm] Add make target build-packages to build the nugets

radical commented:
/azp run runtime-wasm

maraf commented on src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets:
```diff
@@ -394,10 +394,11 @@ Copyright (c) .NET Foundation. All rights reserved.
     </PropertyGroup>
 
-    <ConvertDllsToWebCil Candidates="@(_NewWasmPublishStaticWebAssets)" IntermediateOutputPath="$(IntermediateOutputPath)tmp-webcil\publish" OutputPath="$(_WasmPublishWebCilPath)" IsEnabled="$(_WasmEnableWebcil)">
+    <ConvertDllsToWebCil Candidates="@(_NewWasmPublishStaticWebAssets)" IntermediateOutputPath="$(_WasmPublishTmpWebCilPath)" OutputPath="$(_WasmPublishWebCilPath)" IsEnabled="$(_WasmEnableWebcil)">
```
The task is called twice (build & publish). Should the other place be also updated?

radical replied:
good catch! done.

maraf commented on src/mono/wasm/runtime/loader/polyfills.ts:
```diff
@@ -93,7 +93,8 @@ export async function fetch_like(url: string, init?: RequestInit): Promise<Respo
                     return JSON.parse(read(url, "utf8"));
-                }
+                },
+                text: () => read(url, "utf8")
```
Should we polyfill the `text` function also for node & error case?

radical replied:
done

pavelsavara commented on src/mono/wasm/runtime/loader/assets.ts:
```diff
@@ -168,9 +164,12 @@ export async function mono_download_assets(): Promise<void> {
+                            cleanupAsset(asset);
                         }
+
+                        ++loaderHelpers.actual_downloaded_assets_count;
```
I would prefer to keep the condition
`if (skipBufferByAssetTypes[asset.behavior])`

pavelsavara replied:
the idea is that those *ByAssetTypes collections maintain what should happen, not the code/conditions. And in some cases that's not perfect. Feel free to improve it.

radical committed:
address review feedback from @maraf, and add text implementations for node, and error cases

radical committed:
address review feedback from @maraf

radical committed:
WBT: remove unnecessary code

WithDllImportInMainAssembly: remove `CheckNativeFileLinked` because we
can run the blazor project now.

radical committed:
Merge remote-tracking branch 'origin/main' into wasm-fix-perf

radical committed:
address review feedback from @pavelsavara

radical committed:
Disable failing blazor test on windows
